uploader.go

```diff
@@ -208,10 +241,10 @@ func (u UploadOptions) getBatch(runCtx context.Context) *Batch {
 
 	for {
 		select {
-		case blob, ok := <-u.Input:
+		case inputData, ok := <-u.Input:
 			if ok {
-				b.buffer = append(b.buffer, blob)
-				b.size += uint64(len(blob))
+				b := addToPendingBatch(u, inputData)
+				// keep track of last object type so we know
```

Comment:
If I'm not mistaken, this won't result in all batches being considered done after the batch timeout - just the most recently modified batch. Is that the case?